### DISCUSSION TITLE
Is it possible to temporarily hide some items in the list be setting zero height

### DISCUSSION THREAD

**Discussion by adamvietnam** [petyosi/react-virtuoso#481] [Q&A] [answered]
I'd like to temporarily hide some of the items in a list i.e make some of them invisible and take no space without having to give a the list of different set of items e.g.

<code>item 1: visible</code>
<code>item 2: hidden</code>
<code>item 3: visible</code>
<code>item 4: visible</code>

I've tried adding various things including the following to individual items: 

<code>display: none;</code>

<code>height: 0px;</code>

<code>visibility: hidden;</code>

But I either get screen corruption or, in the case of <code>display:none</code> ,  an exception thrown saying an item can't be zero height.

Does anyone know a way of doing this?

**Comment by petyosi** [comment]
Zero sized items are not supported. A workaround I recommend is a 1px height.

**Comment by adamvietnam** [comment]
Thanks for coming back to me.

I tried that because I have a 5000+ items that in the list ,  1px height
still equates to 500px.

Is there anything else I could try?

**Reply by petyosi** [reply]
Yes, and this is the recommended approach. Filter the items out of your data source rather than hiding them on the UI level.

**Comment by adamvietnam** [comment]
Fair enough - thank you.

**Comment by petyosi** [comment, marked as answer]
Answer - either reduce those items to `1px` height or filter them out on the data level before passing them to the component.

**Comment by adamvietnam** [comment]
Ok great - I appreciate the response.
